app/controllers/api/items_controller.rb

```diff
@@ -0,0 +1,35 @@
+class Api::ItemsController < Api::BaseController
+  def fetch_latest_products
+    page = params[:page].to_i
+    limit = params[:limit].to_i
+    offset = (page - 1) * limit
+
+    query = Item.joins(:collection).select(
```

Comment:
The 'joins' method could lead to performance issues if the 'collections' table grows large. Consider using 'includes' to leverage eager loading and reduce the number of database queries.